service/controller/clusterapi/v18/resources/namespace/current.go

```diff
@@ -2,75 +2,71 @@ package namespace
 
 import (
 	"context"
+	"fmt"
 
 	"github.com/giantswarm/errors/tenant"
 	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/operatorkit/controller/context/reconciliationcanceledcontext"
 	"github.com/giantswarm/operatorkit/controller/context/resourcecanceledcontext"
-	"github.com/giantswarm/tenantcluster"
+	"github.com/giantswarm/operatorkit/example/memcached-operator/logger"
 	corev1 "k8s.io/api/core/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
-	"github.com/giantswarm/cluster-operator/pkg/v18/key"
+	"github.com/giantswarm/cluster-operator/service/controller/clusterapi/v18/controllercontext"
+	"github.com/giantswarm/cluster-operator/service/controller/clusterapi/v18/key"
 )
 
 func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
-	objectMeta, err := r.toClusterObjectMetaFunc(obj)
+	cr, err := key.ToCluster(obj)
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
-
-	// Tenant cluster namespace is not deleted so cancel the resource. The
-	// namespace will be deleted when the tenant cluster resources are deleted.
-	if key.IsDeleted(objectMeta) {
-		r.logger.LogCtx(ctx, "level", "debug", "message", "redirecting namespace deletion to provider operators")
-		resourcecanceledcontext.SetCanceled(ctx)
-		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
-
-		return nil, nil
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return nil, microerror.Mask(err)
 	}
 
-	tenantK8sClient, err := r.getTenantK8sClient(ctx, obj)
-	if tenantcluster.IsTimeout(err) {
-		r.logger.LogCtx(ctx, "level", "debug", "message", "did not get a K8s client for the tenant cluster")
-
-		// We can't continue without a K8s client. We will retry during the
-		// next execution.
-		reconciliationcanceledcontext.SetCanceled(ctx)
-		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling reconciliation")
+	{
+		if cc.Client.TenantCluster.K8s == nil {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "tenant cluster clients not available")
```

Comment:
Is it possible they are not available?

Reply:
Yes, for instance when the cert secret magic is stuck. Then the searcher times out and we got no client.